bagsdb.go

```diff
@@ -112,24 +128,109 @@ func (b *BagsAPI) GetBag(username, bagID string) (BagRecord, error) {
 	var record BagRecord
 	err := b.db.QueryRow(query, bagID, username).Scan(&record.ID, &record.Contents, &record.UserID)
 	if err != nil {
-		return record, err
+		return record, fmt.Errorf("error getting bag id %s for %s: %w", bagID, username, err)
 	}
 	return record, nil
 
 }
 
+func (b *BagsAPI) createDefaultBag(username string) (BagRecord, error) {
+	var (
+		err         error
+		record      BagRecord
+		newBagID    string
+		newContents []byte
+		userID      string
+	)
+	defaultContents := map[string]interface{}{}
+	record.Contents = defaultContents
+
+	if newContents, err = json.Marshal(defaultContents); err != nil {
+		return record, fmt.Errorf("error marshaling default bag: %w", err)
+	}
+
+	if newBagID, err = b.AddBag(username, string(newContents)); err != nil {
+		return record, fmt.Errorf("error adding bag for user %s: %w", username, err)
+	}
+
+	record.ID = newBagID
+
+	if err = b.SetDefaultBag(username, newBagID); err != nil {
+		return record, fmt.Errorf("error setting the default bag for %s: %w", username, err)
+	}
```

Comment:
It might be nice to have support for transactions in this service for cases like this where multiple database updates are being done for a single operation. The consequences of a partial update are minor in this case, so this doesn't have to be done for this PR, but it might be something nice to have in the future.